gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/src/OAuth2Utilities.js

```diff
@@ -17,43 +17,9 @@ var fluid = fluid || require("infusion");
 var gpii = fluid.registerNamespace("gpii");
 fluid.registerNamespace("gpii.oauth2");
 
-gpii.oauth2.errors = fluid.freezeRecursive({
-    missingInput: {
-        msg: "The input field \"%fieldName\" is undefined",
-        statusCode: 400,
-        isError: true
-    },
-    missingDoc: {
-        msg: "The record of %docName is not found",
-        statusCode: 400,
-        isError: true
-    },
-    unauthorizedUser: {
-        msg: "The user %userId is not authorized",
-        statusCode: 401,
-        isError: true
-    },
-    invalidUser: {
-        msg: "Invalid user name and password combination",
-        statusCode: 401,
-        isError: true
-    },
-    unauthorizedAuthCode: {
-        msg: "The authorization code %code is not authorized",
-        statusCode: 401,
-        isError: true
-    },
-    unauthorizedClient: {
-        msg: "The client is not authorized",
-        statusCode: 401,
-        isError: true
-    },
-    unauthorizedAccessToken: {
-        msg: "The access token is not authorized",
-        statusCode: 401,
-        isError: true
-    }
-});
+if (!gpii.oauth2.errors) {
+    require("./OAuth2Const.js");
+}
```

Comment:
Under what circumstances would this test be used? It seems a little unusual. Could we remove it?

Reply:
This js file is used by both browser test scripts and scripts running at the node.js server side. This line is needed for running it at the server side. When running for browser tests, `require()` statement is unrecognized, `OAuth2Const.js` is included via a html file, so the `if` is in place to prevent line 21 to be executed in the browser.

Reply:
But `OAuth2Const.js` is included in

https://github.com/cindyli/universal/blob/GPII-2435/gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/index.js

Wouldn't server side code require that file? And get `OAuth2Const.js` along with `OAuth2Utilities.js`?

Reply:
Re-reading my initial comment, I think I wasn't super-clear. I propose to remove the whole block, not just the guard around the `require`.

Reply:
I see that in the oauth2 code, we are doing similar tests elsewhere:

https://github.com/cindyli/universal/blob/GPII-2435/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStoreUtils.js#L20-L22

I believe that we don't do this in other parts of GPII. But rather, rely on an `index.js` file with requires for a module. This `index.js` file is used in Node.js but not in the browser, where we explicitly include each needed source file.

Reply:
So, for files that are used in both the browser and the server, we minimise the require statements to the essentials, such as

    var fluid = fluid || require("infusion");

And rely on our GPII code dependencies to have been included (either using an `index.js` under Node.js or by an HTML page in the browser case).

Reply:
Makes sense. I've removed those 2 blocks.

Reply:
Awesome, thanks Cindy. Can this one also be removed?

https://github.com/cindyli/universal/blob/GPII-2435/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStore.js#L29-L31

Reply:
Good catch. Removed this one as well.